src/keyfile.c

```diff
@@ -1159,26 +1181,31 @@ void configuration_reload_default_session(void)
 	g_key_file_free(config);
 }
 
-
-gboolean configuration_load(void)
+gboolean read_config_file(gchar const *filename, ConfigPayload payload)
 {
-	gchar *configfile = g_build_filename(app->configdir, "geany.conf", NULL);
+	gchar *configfile = g_build_filename(app->configdir, filename, NULL);
 	GKeyFile *config = g_key_file_new();
 
 	if (! g_file_test(configfile, G_FILE_TEST_IS_REGULAR))
 	{	/* config file does not (yet) exist, so try to load a global config file which may be */
 		/* created by distributors */
 		geany_debug("No user config file found, trying to use global configuration.");
-		SETPTR(configfile, g_build_filename(app->datadir, "geany.conf", NULL));
+		SETPTR(configfile, g_build_filename(app->datadir, filename, NULL));
 	}
 	g_key_file_load_from_file(config, configfile, G_KEY_FILE_NONE, NULL);
 	g_free(configfile);
 
-	load_dialog_prefs(config);
-	load_ui_prefs(config);
-	project_load_prefs(config);
-	configuration_load_session_files(config, TRUE);
-
+	switch (payload)
+	{
+		case PREFERENCES:
+			load_dialog_prefs(config);
+			load_ui_prefs(config);
+			project_load_prefs(config);
+			break;
+		case SESSION:
+			configuration_load_session_files(config, TRUE);
+			break;
+	}
 	/* this signal can be used e.g. to delay building UI elements until settings have been read */
 	g_signal_emit_by_name(geany_object, "load-settings", config);
```

Comment:
The signal should be emitted only once, arguably for the PREFERENCES payload. Might add another signal for load-session ?

Reply:
Added in e406c69f (also fixed the "double vision" document pane bug). I'll leave this unresolved for you to check over the diff.